contracts/truefi2/TrueCreditAgency.sol

```diff
@@ -148,14 +159,19 @@ contract TrueCreditAgency is UpgradeableClaimable {
         require(isPoolAllowed[pool], "TrueCreditAgency: The pool is not whitelisted for borrowing");
         uint8 oldScore = creditScore[pool][msg.sender];
         uint8 newScore = creditOracle.getScore(msg.sender);
+        uint256 currentDebt = borrowed[pool][msg.sender];
+        if (currentDebt == 0) {
+            nextRepaymentTimestamp[pool][msg.sender] = block.timestamp.add(repaymentPeriod);
+        }
 
-        _rebucket(pool, msg.sender, oldScore, newScore, borrowed[pool][msg.sender].add(amount));
+        _rebucket(pool, msg.sender, oldScore, newScore, currentDebt.add(amount));
 
         pool.borrow(amount);
         pool.token().safeTransfer(msg.sender, amount);
     }
 
     function repay(ITrueFiPool2 pool, uint256 amount) external {
+        nextRepaymentTimestamp[pool][msg.sender] = block.timestamp.add(repaymentPeriod);
```

Comment:
If the borrower repays with `amount` less than their interest owed or even 0, this would let them reset their repayment timestamp. We should only reset if `amount >= interest(pool, borrower)`.

Reply:
Borrowers who might not be accustomed to block-by-block interest accrual could get surprised if they send an `amount` a couple cents short of their interest and end up defaulting. In separate PRs we should consider adding `repayInterest(pool)` or `repayInFull(pool)` functions to help them meet our repay schedules.